source/patterns/@aws-solutions-constructs/aws-lambda-kendra/test/lambda-kendra.test.ts

```diff
@@ -0,0 +1,1036 @@
+    kendraDataSourcesProps: [{
+      type: 'S3',
+      dataSourceConfiguration: {
+        s3Configuration: {
+          bucketName: testBucketName,
+        }
+      }
+    }
+    ],
+  });
+
+  const template = Template.fromStack(stack);
+
+  // Check the VPC
+  template.hasResourceProperties("AWS::EC2::VPC", {
+    CidrBlock: "10.0.0.0/16",
+    EnableDnsHostnames: true,
+    EnableDnsSupport: true,
+    InstanceTenancy: "default",
+  });
+
+  // Is the Lambda function associated with the VPC
+  template.hasResourceProperties("AWS::Lambda::Function", {
+    VpcConfig: {
+      SecurityGroupIds: [
+        {
+          "Fn::GetAtt": [
+            "sampleReplaceDefaultSecurityGroupsecuritygroupE5725669",
+            "GroupId"
+          ]
+        }
+      ],
+      SubnetIds: [
+        {
+          Ref: "VpcisolatedSubnet1SubnetE62B1B9B"
+        },
+        {
+          Ref: "VpcisolatedSubnet2Subnet39217055"
+        }
+      ]
+    }
+  });
+  // Check that the Lambda function Policy has proper network access
+  template.hasResourceProperties("AWS::IAM::Policy", {
+    PolicyDocument: {
+      Statement: [
+        {
+          Action: [
+            "ec2:CreateNetworkInterface",
+            "ec2:DescribeNetworkInterfaces",
+            "ec2:DeleteNetworkInterface",
+            "ec2:AssignPrivateIpAddresses",
+            "ec2:UnassignPrivateIpAddresses"
+          ],
+          Effect: "Allow",
+          Resource: "*"
+        },
+        {},
+        {},
+        {}
+      ],
+    },
+    Roles: [
+      {
+        Ref: "sampleLambdaFunctionServiceRole7A3C4AF5"
+      }
+    ]
+  });
+
+  // Check for the Kendra endpoint in the VPC
+  template.hasResourceProperties("AWS::EC2::VPCEndpoint", {
+    ServiceName: {
+      "Fn::Join": [
+        "",
+        [
+          "com.amazonaws.",
+          {
+            Ref: "AWS::Region"
+          },
+          ".kendra"
+        ]
+      ]
+    },
+    VpcId: {
+      Ref: "Vpc8378EB38"
+    },
+    PrivateDnsEnabled: true,
+    SecurityGroupIds: [
+      {
+        "Fn::GetAtt": [
+          "DefaultKENDRAsecuritygroup34536A79",
+          "GroupId"
+        ]
+      }
+    ],
+    SubnetIds: [
+      {
+        Ref: "VpcisolatedSubnet1SubnetE62B1B9B"
+      },
+      {
+        Ref: "VpcisolatedSubnet2Subnet39217055"
+      }
+    ],
+    VpcEndpointType: "Interface"
+  });
+});
+
+test('Launch with existing lambda', () => {
+  const stack = new cdk.Stack();
+  const testTimeout = 3;
+
+  const existingFunction = new lambda.Function(stack, 'existing-function', {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_18_X,
+    handler: 'index.handler',
+    timeout: cdk.Duration.seconds(testTimeout),
+  });
+
+  new LambdaToKendra(stack, 'sample', {
+    existingLambdaObj: existingFunction,
+    kendraDataSourcesProps: [],
+  });
+
+  const template = Template.fromStack(stack);
+  template.resourceCountIs("AWS::Lambda::Function", 1);
```

Comment:
Can we add a `Name` to the existing lambda function, and then assert on that. As is, this test doesn't necessarily prove that a new lambda function wasn't created, since the lambda default timeout is 3 seconds if I recall correctly.